vm-setup/roles/v1aX_integration_test/tasks/verify.yml

```diff
@@ -36,115 +35,72 @@
-
-      - name: Install jq
-        yum:
-          name: jq
-          state: present
-        become: yes
-        become_user: root
-        when:
-          ansible_distribution == "CentOS"
-
-      # Install Calico
-      - name: Download Calico manifest
-        get_url:
-          url: https://docs.projectcalico.org/v3.9/manifests/calico.yaml
-          dest: /tmp/
-          mode: '664'
-        register: calico_manifest
-
-      - name: Replace the POD_CIDR in calico config
-        replace:
-          path: /tmp/calico.yaml
-          regexp: "192.168.0.0/16"
-          replace: "{{ POD_CIDR }}"
-        register: updated_manifest
-
-      - name: Add IP_AUTODETECTION_METHOD in calico config Ubuntu
-        blockinfile:
-          path: /tmp/calico.yaml
-          insertafter: "{{ POD_CIDR }}"
-          block: |
-             # for indentation
-                         - name: IP_AUTODETECTION_METHOD
-                           value: "interface=enp2s0"
-        when:
-          ansible_distribution == "Ubuntu"
-
-      - name: Add IP_AUTODETECTION_METHOD in calico config Centos
-        blockinfile:
-          path: /tmp/calico.yaml
-          insertafter: "{{ POD_CIDR }}"
-          block: |
-             # for indentation
-                         - name: IP_AUTODETECTION_METHOD
-                           value: "interface=eth1"
-        when:
-          ansible_distribution == "CentOS"
-
-      - name: Apply Calico manifest
-        k8s:
-          state: present
-          src: "/tmp/calico.yaml"
-        register: install_cni
-
-      # Check for pods & nodes on the target cluster
-      - name: Check if pods in running state
-        shell: "kubectl get pods -A -o json | jq -r '.items[].status.phase' | grep -v Running"
-        environment:
-          KUBECONFIG: "/home/{{ username }}/.kube/config"
-        retries: 150
-        delay: 20
-        register: running_pods
-        failed_when: >
-          (running_pods.stderr != "") or
-          (running_pods.rc > 1) or
-          (running_pods.stdout != "")
-        until: running_pods.stdout == ""
-
-      - name: Check if nodes in ready state
-        shell: kubectl get nodes | grep -w Ready | wc -l
-        environment:
-          KUBECONFIG: "/home/{{ username }}/.kube/config"
-        retries: 150
-        delay: 3
-        register: ready_nodes
-        failed_when: >
-          (ready_nodes.stderr != "") or
-          (ready_nodes.rc != 0) or
-          (ready_nodes.stdout != NUMBER_OF_BMH)
-        until: ready_nodes.stdout == NUMBER_OF_BMH
\ No newline at end of file
+  - name: Fetch target cluster kubeconfig
+    shell: "kubectl get secrets {{ CLUSTER_NAME }}-kubeconfig -n {{ NAMESPACE }} -o json | jq -r '.data.value'| base64 -d > /tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+
+  # Install Calico
+  - name: Download Calico manifest
+    get_url:
+      url: https://docs.projectcalico.org/v3.9/manifests/calico.yaml
+      dest: /tmp/
```

Comment:
maybe you should make sure the name includes the cluster name for example, to prevent issues with parallel runs on the same machine

Reply:
cluster name on the name of the file?

Reply:
I think it is okay for now since we don't yet have a case where we have parallel runs as you mentioned.

Reply:
that's true, but what about successive runs ? https://docs.ansible.com/ansible/latest/modules/get_url_module.html#parameter-force that's applicable for feature tests

Reply:
hum, since you gave dest as a directory, it looks like you do not need force : https://docs.ansible.com/ansible/latest/modules/get_url_module.html#parameter-dest so that is ok as is